File: src/core/data_formats/parsing/quest/Episode.ts
export enum Episode {
    I = 1,
    II = 2,
    IV = 4,
}

export const EPISODES = [Episode.I, Episode.II, Episode.IV];

export function check_episode(episode: Episode): void {
    if (Episode[episode] == undefined) {
        throw new Error(`Invalid episode ${episode}.`);
    }
}

File: lib/src/commonMain/kotlin/world/phantasmal/lib/fileformats/quest/Episode.kt
package world.phantasmal.lib.fileformats.quest

enum class Episode {
    I,
    II,
    IV;

    companion object {
        fun fromInt(episode: Int) = when (episode) {
            1 -> I
            2 -> II
            4 -> IV
            else -> error("$episode is invalid.")
        }
    }
}
